src/main/java/org/broadinstitute/hellbender/engine/spark/SparkSharder.java

```diff
@@ -283,11 +284,13 @@ public T apply(@Nullable Tuple2<I, Iterable<L>> input) {
             int partitionIndex = splitPoint.getPartitionIndex();
```

Comment:
Can you add unit tests for `computePartitionReadExtents()` to trigger the new error conditions?